src/streaming-services/nrk/NrkApi.ts

```diff
@@ -65,23 +129,29 @@ class _NrkApi extends Api {
 			}
 			let isLastPage = false;
 			let items: Item[] = [];
-			const historyItems: NrkHistoryItem[] = [];
+			const historyItems: NrkProgressItem[] = [];
 			do {
 				const responseText = await Requests.send({
-					url: `${this.HISTORY_API_URL}?pg=${nextPage}`, //TODO figure out if pagination is even supported in the API
+					url: this.HISTORY_API_URL,
 					method: 'GET',
+					headers: {
+						Authorization: 'Bearer ' + this.token,
+					},
 				});
-				const responseJson = JSON.parse(responseText) as NrkHistoryItem[];
-				if (responseJson && responseJson.length > 0) {
-					itemsToLoad -= responseJson.length;
-					historyItems.push(...responseJson);
+				const responseJson = JSON.parse(responseText) as NrkProgressResponse;
+				const { progresses } = responseJson;
+				this.HISTORY_API_URL = this.API_HOST_URL + responseJson._links.next.href;
```

Comment:
This actually leads to an error for me. I guess because my account is new and I only have 2 items in my history, `responseJson._links.next` isn't available in the response.